packages/react-core/src/components/Timestamp/__tests__/Timestamp.test.tsx

```diff
@@ -0,0 +1,196 @@
+
+  expect(screen.getByText('1/1/2022, 12:00:00 AM')).toBeInTheDocument();
+});
+
+test('Renders with custom formatting when dateFormat and timeFormat are passed in', () => {
+  render(
+    <Timestamp
+      datetime={new Date(2022, 0, 1).toISOString()}
+      date={new Date(2022, 0, 1)}
+      dateFormat={TimestampFormat.full}
+      timeFormat={TimestampFormat.short}
+    />
+  );
+
+  expect(screen.getByText('Saturday, January 1, 2022 at 12:00 AM')).toBeInTheDocument();
+});
+
+test('Renders with only date when dateFormat is passed in', () => {
+  render(
+    <Timestamp
+      datetime={new Date(2022, 0, 1).toISOString()}
+      date={new Date(2022, 0, 1)}
+      dateFormat={TimestampFormat.full}
+    />
+  );
+
+  expect(screen.getByText('Saturday, January 1, 2022')).toBeInTheDocument();
+});
+
+test('Renders with only time when timeFormat is passed in', () => {
+  render(
+    <Timestamp
+      datetime={new Date(2022, 0, 1).toISOString()}
+      date={new Date(2022, 0, 1)}
+      timeFormat={TimestampFormat.short}
+    />
+  );
+
+  expect(screen.getByText('12:00 AM')).toBeInTheDocument();
+});
+
+test('Renders with custom formatting when customFormat is passed in', () => {
+  render(
+    <Timestamp
+      datetime={new Date(2022, 0, 1).toISOString()}
+      date={new Date(2022, 0, 1)}
+      customFormat={{ year: '2-digit', month: 'short', weekday: 'short', day: 'numeric', hour: 'numeric' }}
+    />
+  );
+
+  expect(screen.getByText('Sat, Jan 1, 22, 12 AM')).toBeInTheDocument();
+});
+
+test('Renders with custom content', () => {
+  render(
+    <Timestamp datetime={new Date(2022, 0, 1).toISOString()} date={new Date(2022, 0, 1)}>
+      2 days ago
+    </Timestamp>
+  );
+
+  expect(screen.getByText('2 days ago')).toBeInTheDocument();
+});
+
+test('Renders with a custom suffix when displaySuffix is passed in', () => {
+  render(
+    <Timestamp datetime={new Date(2022, 0, 1).toISOString()} date={new Date(2022, 0, 1)} displaySuffix="US Eastern" />
+  );
+
+  expect(screen.getByText('1/1/2022, 12:00:00 AM US Eastern')).toBeInTheDocument();
+});
+
+test('Renders with 12 hour time by default for US locale', () => {
+  render(<Timestamp datetime={new Date(2022, 0, 1, 13, 0).toISOString()} date={new Date(2022, 0, 1, 13, 0)} />);
+
+  expect(screen.getByText('1/1/2022, 1:00:00 PM')).toBeInTheDocument();
+});
+
+test('Renders with 12 hour time when is12Hour is passed in for US locale', () => {
+  render(
+    <Timestamp datetime={new Date(2022, 0, 1, 13, 0).toISOString()} date={new Date(2022, 0, 1, 13, 0)} is12Hour />
+  );
+
+  expect(screen.getByText('1/1/2022, 1:00:00 PM')).toBeInTheDocument();
+});
+
+test('Renders with 24 hour time when is12Hour is set to false for US locale', () => {
+  render(
+    <Timestamp
+      datetime={new Date(2022, 0, 1, 13, 0).toISOString()}
+      date={new Date(2022, 0, 1, 13, 0)}
+      is12Hour={false}
+    />
+  );
```

Comment:
Should these tests be using the `locale` prop?

Reply:
Also could you add a test for the `locale` prop itself?

Reply:
Ah good catch. Updated these tests and added some tests for the `locale` prop!